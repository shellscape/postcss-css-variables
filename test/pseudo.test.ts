import { fixture, process } from './helpers';

const tests = {
  'should work with pseudo selectors': 'pseudo-selector',
  'should work with variables declared in pseudo selectors': 'pseudo-selector-declare-variable'
};

describe('pseudo', async () => {
  test.each(Object.entries(tests))('%s', async (_desc, file) => {
    const css = await fixture(`pseudo/${file}`);
    const result = await process(css);
    expect(result).toMatchSnapshot();
  });
});
