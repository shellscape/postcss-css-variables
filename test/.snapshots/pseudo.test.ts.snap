// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pseudo > should work with pseudo selectors 1`] = `".foo{color:#ff0000}.foo:before{color:#ff0000}"`;

exports[`pseudo > should work with variables declared in pseudo selectors 1`] = `
".foo{color:#ff0000}
.foo:hover{color:#00ff00}"
`;
